### PR TITLE
Add 'mark thread as done' functionality

### PR DESCRIPTION
**Pull request by danhcole**
Fixes https://github.com/google/go-github/issues/3264.

### CODE REVIEW

File: github/activity_notifications.go
Comment (gmlewis) [inline github/activity_notifications.go:187]: According to this: https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#mark-a-thread-as-done
`id` is an integer, so let's please change this:

```suggestion
func (s *ActivityService) MarkThreadDone(ctx context.Context, id int64) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @danhcole !
One minor tweak, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/activity_notifications.go
Comment (danhcole) [reply github/activity_notifications.go:187]: Sounds good to me, I'll update the PR.  The MarkThreadRead function (which I shamelessly copied from) has the same issue, would you like me to fix it as part of this PR, or put in a separate PR to keep things clean?

Comment (gmlewis) [reply github/activity_notifications.go:187]: Yes, a separate PR would be great since that will be a breaking change. Thank you!

File: github/activity_notifications_test.go
Comment (gmlewis) [inline github/activity_notifications_test.go:221]: These strings will need to be changed to `int64` for tests to pass.

Comment (danhcole) [reply github/activity_notifications_test.go]: Whops, fixed and tests pass locally now.

**Review by gmlewis — APPROVED** [review]
Thank you, @danhcole !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin!
Merging.

**Pull request merged**
